src/H5FDs3comms.c

```diff
@@ -921,21 +947,23 @@ H5FD__s3comms_s3r_configure_aws(s3r_t *handle, const H5FD_ros3_fapl_t *fa, const
  *----------------------------------------------------------------------------
  */
 herr_t
-H5FD__s3comms_s3r_read(s3r_t *handle, haddr_t offset, size_t len, void *dest)
+H5FD__s3comms_s3r_read(s3r_t *handle, haddr_t offset, size_t len, void *dest, size_t dest_size)
```

Comment:
With `CURLOPT_FAILONERROR` turned off, the data that comes back from a request on failure might be an additional response body rather than the bytes being looked for. In this case, it's possible that the previous logic could have tried to put more bytes into the destination buffer than what's available, so a buffer size parameter was added here. The ROS3 VFD needs to be redesigned a bit to correctly handle these situations, but this is still a useful check anyway.